coilsnake/modules/eb/TitleScreenModule.py

```diff
@@ -0,0 +1,641 @@
+            image.save(f)
+
+        # Write out the background's animated frames
+        for frame in xrange(NUM_ANIM_FRAMES):
+            palette = EbPalette(NUM_SUBPALETTES, BG_SUBPALETTE_LENGTH)
+            if frame < CHARS_NUM_ANIM_SUBPALETTES:
+                palette[0, CHARS_ANIM_SLICE] = \
+                    self.chars_anim_palette.get_subpalette(frame)[0, :]
+            else:
+                palette[0, :] = self.bg_palette.get_subpalette(0)[0, :]
+                palette[0, BG_ANIM_SLICE] = \
+                    self.bg_anim_palette.get_subpalette(
+                        frame - CHARS_NUM_ANIM_SUBPALETTES
+                    )[0, :]
+            with resource_open(BG_FRAMES_PATH.format(frame), "png") as f:
+                image = self.bg_arrangement.image(self.bg_tileset, palette)
+                image.save(f)
+
+    def write_chars_data_to_project(self, resource_open):
+        # Build an arrangement combining every character for convenience
+        chars_positions = {}
+        arrangement = EbTileArrangement(3*9, 6)
+        for c, layout in enumerate(self.chars_layouts):
+            top_left = {'x': 128, 'y': 128}
+            for e, entry in enumerate(layout):
+                tile = entry.tile & (CHARS_NUM_TILES - 1)
+                top_left['x'] = min(top_left['x'], int(entry.x))
+                top_left['y'] = min(top_left['y'], int(entry.y))
+                x = c*3 + (entry.x+16)/8
+                y = (entry.y+24)/8
+                arrangement[x, y].tile = tile
+                if not entry.is_single():
+                    arrangement[x+1, y].tile = tile + 1
+                    arrangement[x, y+1].tile = tile + 16
+                    arrangement[x+1, y+1].tile = tile + 17
+            chars_positions[c] = {
+                'x': c*3*8,
+                'y': 0,
+                'width': 3*8,
+                'height': 6*8,
+                'top_left_offset': top_left,
+                'unknown': layout[0].unknown
+            }
+
+        # Write the characters animation frames
+        for p in xrange(CHARS_NUM_ANIM_SUBPALETTES):
+            with resource_open(CHARS_FRAMES_PATH.format(p), "png") as f:
+                image = arrangement.image(
+                    self.chars_tileset,
+                    self.chars_anim_palette.get_subpalette(p)
+                )
+                image.save(f)
+
+        # Write out the initial characters palette
+        with resource_open(CHARS_INITIAL_PATH, "png") as f:
+            image = arrangement.image(
+                self.chars_tileset,
+                self.chars_palette
+            )
+            image.save(f)
+
+        # Write out the positions of the characters
+        with resource_open(CHARS_POSITIONS_PATH, "yml") as f:
+            yml_dump(chars_positions, f, False)
+
+    def upgrade_project(
+            self, old_version, new_version, rom, resource_open_r,
+            resource_open_w, resource_delete):
+        if old_version < 9:
```

Comment:
You should add version 9 to `project.py` in `VERSION_NAMES` and set it as the `FORMAT_VERSION`